homeassistant/components/rest/binary_sensor.py

```diff
@@ -114,16 +124,72 @@ def is_on(self):
         """Return true if the binary sensor is on."""
         return self._is_on
 
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return self._attr_extra_state_attributes
+
     def _update_from_rest_data(self):
         """Update state from the rest data."""
-        if self.rest.data is None:
+        response = self.rest.data
+        _LOGGER.debug("Data fetched from resource: %s", response)
+
+        if response is None:
             self._is_on = False
 
-        response = self.rest.data
+        if self.rest.headers is not None:
+            # If the http request failed, headers will be None
+            content_type = self.rest.headers.get("content-type")
+
+            if content_type and (
+                content_type.startswith("text/xml")
+                or content_type.startswith("application/xml")
+                or content_type.startswith("application/xhtml+xml")
+                or content_type.startswith("application/rss+xml")
```

Comment:
`startswith` argument can be a tuple, so no need for all these call to `startwith`

https://docs.python.org/3/library/stdtypes.html#str.startswith

Reply:
I would make a constant and import it in both `binary_sensor.py` and `sensor.py` (the latter in a separate PR).

```python
MIME_TYPES = (
    "text/xml",
    "application/xml",
    "application/xhtml+xml",
    "application/rss+xml",
)
````

Is that ok - or would you rather duplicate the tuple in both files?

Reply:
or maybe use python's `mimetypes` ?

Reply:
I am not really able to utilize `mimetypes` in a helpful way in this case - I was not able to initialize an empty MimeTypes database (if I understand correctly, one  would need to initialize it and then empty the dictionaries?). Any help is very welcome.

Reply:
```python
import mimetypes

mimetypes.init()

mimes = [mimetypes.types_map[_type] for _type in ['.rss', '.atom', '.xml']]
```

Reply:
Is it bad practice to have this in `const.py`?

Reply:
completely acceptable :)

Reply:
Great - thanks! Has been implemented.

Reply:
I have realized that we are now missing `text/xml` (which is not in included in `mimetypes.types_map`) -- from a quick survey it seems that the usual type is anyways `application/xml`, yet I have read mixed opinions. Is that a problem?

Shall we just drop `text/xml`, or manually inject it to the tuple?
(Sorry for all the noise)